File: MusicPlayer/app/src/main/java/com/zukron/musicplayer/adapter/LibraryAdapter.java
package com.zukron.musicplayer.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.card.MaterialCardView;
import com.zukron.musicplayer.R;
import com.zukron.musicplayer.model.Library;

import java.util.ArrayList;

/**
 * Project name is Music Player
 * Created by Zukron Alviandy R on 7/27/2020
 */
public class LibraryAdapter extends RecyclerView.Adapter<LibraryAdapter.ViewHolder> {
    private Context context;
    private ArrayList<Library> libraries;
    private OnSelected onSelected;

    public LibraryAdapter(Context context, ArrayList<Library> libraries) {
        this.context = context;
        this.libraries = libraries;
    }

    public void setOnSelected(OnSelected onSelected) {
        this.onSelected = onSelected;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        View view = layoutInflater.inflate(R.layout.item_library, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        final Library library = libraries.get(position);

        holder.tvTitleItemLibrary.setText(library.getTitle());
        holder.mcvItemLibrary.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onSelected.onSelectedItem(library);
            }
        });
    }

    @Override
    public int getItemCount() {
        return libraries.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private MaterialCardView mcvItemLibrary;
        private TextView tvTitleItemLibrary;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            mcvItemLibrary = itemView.findViewById(R.id.mcv_item_library);
            tvTitleItemLibrary = itemView.findViewById(R.id.tv_title_item_library);
        }
    }

    public interface OnSelected{
        void onSelectedItem(Library library);
    }
}

File: MusicPlayer/app/src/main/java/com/zukron/musicplayer/adapter/LibraryAdapter.kt
package com.zukron.musicplayer.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.card.MaterialCardView
import com.zukron.musicplayer.R
import com.zukron.musicplayer.model.Library

/**
 * Project name is Music Player
 * Created by Zukron Alviandy R on 9/3/2020
 * Contact me if any issues on zukronalviandy@gmail.com
 */
class LibraryAdapter(private val context: Context) :
        RecyclerView.Adapter<LibraryAdapter.ViewHolder>() {

    var libraries = mutableListOf<Library>()
    var onSelected: OnSelected? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(context)
        val view = layoutInflater.inflate(R.layout.item_library, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val library = libraries[position]

        holder.tvTitle.text = library.title
        holder.materialCard.setOnClickListener {
            onSelected.let {
                it!!.onSelectedItem(library)
            }
        }
    }

    override fun getItemCount(): Int {
        return libraries.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val materialCard = itemView.findViewById<MaterialCardView>(R.id.material_card_view)
        val tvTitle = itemView.findViewById<TextView>(R.id.tv_title)
    }

    interface OnSelected {
        fun onSelectedItem(library: Library)
    }
}